### PR TITLE
Search page

### PR DESCRIPTION
**Pull request by camillaume**

### CODE REVIEW

**Review by ghost — DISMISSED** [review]

**Review by LuigiMenez — DISMISSED** [review]
Ok for me.

File: frontend/src/pages/Search/index.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    search: "",
+  });
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    if (!formData.search) {
+      setSearchResults([]);
+    } else {
+      axios
+        .get(`http://localhost:5000/searchs?needle=${formData.search}`)
```
Comment (LoicBrassart) [inline frontend/src/pages/Search/index.jsx]: Put this `http://localhost:5000` in your `.env` file

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Put your backend url in your .env file to ease up deployment later

**Review by ghost — DISMISSED** [review]

**Review by LuigiMenez — DISMISSED** [review]
Ok for me.

**Review by LuigiMenez — DISMISSED** [review]
Ok for me.

**Review by ghost — DISMISSED** [review]

**Review by LuigiMenez — DISMISSED** [review]
Ok for me.

File: backend/src/router.js [hunk 2]
```diff
@@ -3,12 +3,16 @@ const express = require("express");
...
+  MusicStyleController,
   ArtistController,
+  SearchableController,
 } = require("./controllers");
 
 const router = express.Router();
 
 router.get("/items", ItemController.browse);
+router.get("/musicStyles", MusicStyleController.browse);
+router.get("/artists", ArtistController.browse);
```
Comment (LoicBrassart) [inline backend/src/router.js]: - not mix /items routes with useful ones
- delete /items routes (only useful as a template)
- use lowercase route names

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -1,18 +1,22 @@
 import Navbar from "@components/Navbar";
 import { Routes, Route } from "react-router-dom";
 import Profile from "@pages/Profile";
+import Search from "@pages/Search";
+import "./App.css";
```
Comment (LoicBrassart) [inline frontend/src/App.jsx]: No CSS

File: frontend/src/App.jsx [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
-    <div className="App">
-       <Navbar />
-       <Routes>
+   <div className="App">
+    <Navbar />
+    <Routes>
+      <Route path="/profile" element={<Profile />} />
+      <Route path="/search" element={<Search />} />
       <Route path="/" element={<Home />} />
+      <Route path="/profile" element={<Profile />} />
```
Comment (LoicBrassart) [inline frontend/src/App.jsx:14]: This route is declared THREE lines above ... Please review carefully

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Please be careful when reviewing PRs

**Review by ghost — DISMISSED** [review]
fix all change requiered

**Review by ghost — APPROVED** [review]

**Review by LuigiMenez — APPROVED** [review]
Ok for me.

**Comment by LoicBrassart** [comment]
@SDecambray Please be careful about your commit messages, I have no idea of what you did by reading them...

**Review by LoicBrassart — APPROVED** [review]
No more Evil

**Pull request merged**